### PR TITLE
[RA] Rename and alter the 'Hind' to 'Cobra'

### PR DESCRIPTION
**Pull request by SoScared**
following https://github.com/OpenRA/OpenRA/issues/12197#issuecomment-281485039

cobra 0000

hind-to-cobra6

As for the hind.shp:

- Cut off the Hind's "wings", turned dual chainguns into single chaingun (added turret).
- Grey-scaled and slightly merged the blue tinted bubble canopies.

Also altered the chaingun min. range from 0c768 to 1c768 due to a new weapons offset. Balance and performance wise I don't see any potential hazard. As far as I'm concerned the Hind name in the .shp and .yaml ought to just stay given the original unit being the foundation for the Cobra.

alter3

video footage: https://www.youtube.com/watch?v=F9uSI2N6ZSs

### CODE REVIEW

**Review by Micr0Bit — APPROVED** [review]
tested and works , this will do the job good enough

**Comment by AoAGeneral** [comment]
Is it correct that this unit now has a 12 ammo pool instead of 24 and a single firing mechanic? O.o does this mean it does less damage and has to reload more often then originally?

**Comment by pchote** [comment]
It's gone from 2 guns (12 ammo each) doing 30 damage per hit down to 1 gun (12 ammo total) doing 60 damage per hit.

**Comment by SoScared** [comment]
Interesting. Does this affect the reload time on the Helipad?

**Comment by SoScared** [comment]
Yup

**Comment by SoScared** [comment]
Doubled Ammopool ReloadDelay to compensate.

**Comment by SoScared** [comment]
@AoAGeneral: addressing the performance discussion on #IRC, would be nice with some playtesting mass cobras vs mass infantry close quarters.

**Comment by AoAGeneral** [comment]
I would be willing to have a look at that. Unfortunately im more of the dumb rock and don't know how to test forked code. :/

**Comment by Smittytron** [comment]
:-1: for appeasement.

**Comment by ScottNZ** [comment]
I think this should be implemented as a new unit, in case there is a RA mission which relies on the Soviets actually having hinds. It would be just as silly as it is now to see the Soviet AI flying Cobras around.

**Comment by atlimit8** [comment]
Multiplayer balance causes a lot of discourse. The Hind is Russian and the Cobra is American. Adding the Cobra to fix the Allies having the Hind works. Making the Russians use the Cobra in missions does not. It would be nice to get all of the missions from the original. Not that they have to be exact.

By the way, the Cobra is from the 60s and the Hind from the 70s. However, Einstein invented time travel and the Allies won the space race with their satellites.

Adding `PR: Changes requested` since a few of us agree that `HIND` should not be Cobra.

**Comment by MustaphaTR** [comment]
About hind itself. I think we should use it to replace Soviet Chinooks for missions that they had one.

**Comment by SoScared** [comment]
Makes sense avoiding RA missions playing with Cobras. Should be a relatively easy fix.

**Comment by SoScared** [comment]
> @Smittytron: :-1: for appeasement.

Elaborate please.

**Comment by Smittytron** [comment]
> Elaborate please.

Appeasement is making a change to keep people happy without any other motivations. Funny thing is, the guy who started this kerfuffle saw this pr and still isn’t satisfied: http://www.sleipnirstuff.com/forum/viewtopic.php?f=82&t=19980

When I said history buffs/military junkies are still going to complain, I said it because I am one. I know how these people think. (The Mig from the origional [videos](https://www.youtube.com/watch?v=sNG8xmuHIHw) was clearly a Mig-29 but the in game it only has one tail-fin! Gaaaaaaah fix it!!)

The ticket and thread went up Monday, the thread was downvoted to oblivion by Tuesday and a pr pops up Wednesday. 

The first development goal on the GitHub OpenRA wiki is this:
1.	Recreating the original 2D Command & Conquer games with enhancements inspired by more modern RTS games. OpenRA should retain the feel and nostalgia of the older games, but remove the frustrating and dated elements.

The way I see it, making a visual change here violates the above goal because such a change brings OpenRA further away from the source without enhancing gameplay in any way.

**Comment by SoScared** [comment]
@Smittytron: I appreciate your concern but it's misplaced. Its true that the discussion beforehand triggered me to write this PR but the idea of renaming Hind to an Allied Heli is an old idea consistently brought up by new players as far as I can personally remember - which have to be about 6 years now.

Make no mistake I don't start PR's to appease noone and I never have. This PR is what _I_ want. I don't value polls, I definitely don't value adversarial opinion spamming as done with the linked discussion thread and I don't value majority opinions unless its from a group of people I respect and feel confident in that they know what they're doing.

As for the change with the .shp unit and icon in relation to originality I can be more sympathetic but in this case I consider this matter to be a non-issue. It's a limited alteration to effectively pick off a long-standing mole on the RA mod's back. I don't worry about the "slippery-slope" either - there's no other RA mod related issue that comes anywhere close to the 'rename Hind' demand.

Ego bruising as a result of one rouge person pushing his or her opinions should be left alone to the forums, not pull the direction of a PR. Allowing that would be just as bad as any appeasement.

**Comment by Smittytron** [comment]
Here's my counter proposal:
Place the following text in the [FAQ](https://github.com/OpenRA/OpenRA/wiki/FAQ) section under the title, 'The Hind is a Soviet aircraft. Why is it an Allied unit?' ...


'In the OpenRA timeline: As Stalin became more aggressive with his plans for Europe, Soviet aerospace engineer Mikhail Mil became disillusioned with the Communist cause and defected to the Allies with the plans for the Hind attack helicopter.

In order to balance multiplayer the Hind was transferred to the Allies. The Hind has since become a crucial element of the Allied midgame arsenal, providing scouting and crowd-control vs. infantry.

While yes, we are aware that the Hind is a real-life Soviet-era vehicle, and yes, it would be possible to visually remove the Hind and add an Allied helicopter like the AH-1 Cobra, visually changing the unit would take OpenRA further away from the original Red Alert. 

As stated in our development goals, OpenRA should retain the feel and nostalgia of the older games. Because a visual change adds nothing to gameplay, and because the Hind is a unit in the original game, we have no plans to remove the Hind from the Allies at this time.'


Simply link the FAQ to anyone that brings up the issue again.
I believe zinc mentioned the possibility of a making mission based around the Allies stealing the plans for the Hind. I don't believe it's necessary but if there's an interested party it could compliment my idea.

**Comment by SoScared** [comment]
Sounds like a bloated solution. Also equipping ourselves with cheap FAQ links to address every newcomer with a legitimate design question would likely be more like adding fuel to the fire.

Wherever the RA mod has altered the original structure it's easily explained in terms of balance. The Hind suffers uniquely by carrying the name of a Soviet unit.

**Comment by pchote** [comment]
I agree strongly with all the points @SoScared mentioned in his last two comments.

**Comment by FrameLimiter** [comment]
Pretty minor issue, but the Cobra does more damage vs Heavy/Concrete targets than the Hind does. (1HP more per shot; 12HP more per full payload)

This is because of the rounding down of fractions: (60dmg - 75% = 15dmg) (30dmg - 75% = 7.5dmg)

Setting the Heavy & Concrete modifier to 24 will keep the damage the same:
(60dmg - 76% = 14.4dmg)

**Comment by SoScared** [comment]
@FrameLimiter thanks! Adding it for consistency sake.

**Comment by SoScared** [comment]
Exposed the altered hind/hindicon .shp as cobra/cobraicon .shp in order to retain the original assets.

File: mods/ra/sequences/aircraft.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ heli:
 	icon: heliicon
 
 hind:
```
Comment (atlimit8) [inline mods/ra/sequences/aircraft.yaml]: Please rename this `cobra`

**Review by pchote — COMMENTED** [review]
We have conflicting requests here, because I had specifically asked @SoScared to keep this on HIND.

The logic here is that this is repurposing the same unit for a new role (just like the thief/hijacker and SSM launcher/mobile SAM), and making this a separate unit would be inviting a repeat of this godawful episode in 6 months when somebody insists on bringing back HIND as a unit for a new soviet faction.

**Comment by atlimit8** [comment]
Don;t forget renaming `HIND.Husk` to `COBRA.Husk`.

You should probably add this as well then:
```yaml
HIND:
	Inherits: COBRA
```
Unless you reverse it:
```yaml
COBRA:
	Inherits: HIND
	Buildable:
		Description: Helicopter gunship armed\nwith a powerful chaingun.\n  Strong vs Infantry, Light armor.\n  Weak vs Tanks, Aircraft
	Tooltip:
		Name: Cobra
	Armament@PRIMARY:
		LocalOffset: 875,0,-225
	-Armament@SECONDARY:
	AmmoPool:
		Ammo: 12
		ReloadDelay: 16
```

**Comment by atlimit8** [comment]
It might be a good idea to keep the `HIND` rules somewhere for missions to use.

**Comment by atlimit8** [comment]
Oops, Narrowing objective.

It might be a good idea to have something like this for missions:

In  `mods/ra/rules/soviet-hind.yaml`:
```yaml
SOVHIND:
	Inherits: COBRA
	Buildable:
		Description: Helicopter gunship armed\nwith dual chainguns.\n  Strong vs Infantry, Light armor.\n  Weak vs Tanks, Aircraft
	Tooltip:
		Name: Hind
	Armament@PRIMARY:
		Weapon: HindChainGun
		LocalOffset: 85,-213,-85
	Armament@SECONDARY:
		Name: secondary
		Weapon: HindChainGun
		LocalOffset: 85,213,-85
		MuzzleSequence: muzzle
	AmmoPool:
		Ammo: 24
		ReloadDelay: 16
```

In `mods/ra/sequences/aircraft.yaml`:
```yaml
 sovhind:
	idle: hind
		Facings: 32
		UseClassicFacingFudge: True
	rotor: lrotorlg
		Length: 4
	slow-rotor: lrotorlg
		Start: 4
		Length: 8
	muzzle: minigun
		Length: 6
		Facings: 8
	icon: hindicon
```
In `mods/ra/weapons/smallcaliber.yaml`:
```yaml
HindChainGun:
	Inherits: ChainGun
	MinRange: 0c768
 	Warhead@1Dam: SpreadDamage
		-Damage: 60
```

File: mods/ra/sequences/aircraft.yaml (hunk 1 above)
Comment (Mailaender) [inline mods/ra/sequences/aircraft.yaml]: It would then need a lot more changes to the legacy map import and a map update rule.

**Comment by FiveAces** [comment]
I prefer this option over the other Cobra that was proposed. It is less intrusive than the other pull request.
One minor thing about the balance though: From what I can gather in your video, it seems like replacing the stub wing chaingun armaments with a nose-mounted chaingun has slightly increased minimum range (I just tested the release Hind in-game and to me it seems like it has increased from .5 cells to an entire cell).
Not a big deal, though. This to me is clearly the better of the two options. +1 from me!

**Comment by SoScared** [comment]
@FiveAces yeah the +1 Min.Range was done purely to match the altered fire animation to avoid making close kills look awkward. Unlikely to cause trouble IMO but it hasn't been mapped out. Soon there'll be a playtest for you to throw around your famous air blobs :)

**Comment by SoScared** [comment]
@pchote @atlimit8 what's the score on the above suggestions?

**Comment by pchote** [comment]
I still prefer that if (and only if) a mission requires hinds that they can override the `HIND` definition back to its original state.  We don't want missions to use both hinds and cobras, and so shouldn't make it easy for people to do that (making sure that it isn't impossible, which you've now done, is necessary and sufficient).

**Comment by pchote** [comment]
@SoScared: @FiveAces was referring to the *maximum* range of the weapon, which has been effectively increased by moving the muzzle half a cell closer to the front of the unit.

**Comment by SoScared** [comment]
Looked to me like 5A was talking about the minimum range above. Anyways an increase of max. range would be unacceptable. I'll look into it.

**Comment by atlimit8** [comment]
I just don't want the rules lost; so, I will settle for this.

**Comment by SamKeightley** [comment]
@FiveAces I disagree with your thoughts on your perceived levels on intrusiveness. There's nothing stopping the re-implementation of Hinds at a later date if they are required. The act of doing so would be a 2 minute job. Also my solution is not a Cobra..

@atlimit8 You seem to misunderstand, there would be no rules lost in either solution, both solutions are purely aesthetic changes with minor rules edits.

**Comment by atlimit8** [comment]
@SamKeightley:
1. Any rule change is intrinsically removal of its old form. I was hoping that the mission makers would not need to reconstruct the current form of the rules.
2. They are not purely aesthetic, 2 guns -> 1 front mounted with twice the firepower.

**Comment by pchote** [comment]
The tone of the discussions around the hind/cobra/huey topic across Github/forums/IRC has been downright disgusting, and at this point it seems clear that merging *any* of these changes will only continue the toxic drama.

The Allied Hind will have to stay.

**Comment by atlimit8** [comment]
It is unfortunate that nothing will come of all this work; at least for the time being.

**Comment by Mailaender** [comment]
It can still be used in custom maps on http://resource.openra.net/ so nothing is wasted.

**Comment by atlimit8** [comment]
Perhaps custom maps on http://resource.openra.net/ could be used later on for testing 'Hind' replacements after their has been some time to cool down.

**Pull request closed without merging**